windows/process.mpl

```diff
@@ -1,159 +1,143 @@
+  processSecurityAttributes: SECURITY_ATTRIBUTES Cref; # We want it to be Nil by default
+  threadSecurityAttributes:  SECURITY_ATTRIBUTES Cref; #
 
-Process: [{
-  private startupInfo: STARTUPINFOW;
-  private processInfo: PROCESS_INFORMATION;
+  init: [startupInfo storageSize Nat32 cast @startupInfo.!cb];
 
   start: [
-    command: toUtf16;
-    result: processInfo storageAddress
-      startupInfo storageAddress
+    command:;
+    result:
+      @processInformation
+      startupInfo
       0nx
       0nx
       0n32
       0
-      0nx
-      0nx
+      threadSecurityAttributes
+      processSecurityAttributes
       command.data storageAddress
       0nx
-      CreateProcessW;
+      CreateProcessW 0 = ~
+    ;
 
-    result 0 "CreateProcessW" getErrorMessage
+    result
   ];
 
-  wait: [
-    result: INFINITE processInfo.hProcess WaitForSingleObject;
-    result WAIT_FAILED "WaitForSingleObject" getErrorMessage
-  ];
+  tillActive: [INFINITE processInformation.hProcess WaitForSingleObject WAIT_OBJECT_0 =];
 
-  terminate: [
-    exitCode:;
-    result: exitCode processInfo.hProcess TerminateProcess;
-    result 0 "TerminateProcess" getErrorMessage
+  close: [
+    processInformation.hProcess CloseHandle 0 = ~ [processInformation.hThread CloseHandle 0 = ~] &&
   ];
 
-  getExitCode: [
-    out: {
-      exitCode: Nat32;
-      error:    String;
-    };
-
-    result:
-      out.exitCode storageAddress
-      processInfo.hProcess
-      GetExitCodeProcess;
-
-    result 0 "GetExitCodeProcess" getErrorMessage @out.!error
-
-    @out
-  ];
-
-  private getErrorMessage: [
-    code: failCode: functionName:;;;
-    code failCode = ~ [String] [
-      (functionName " failed, error code: " GetLastError) assembleString
-    ] if
-  ];
-
-  private DIE: [
-    (
-      processInfo.hProcess
-      processInfo.hThread
-      startupInfo.hStdInput
-      startupInfo.hStdOutput
-      startupInfo.hStdError
-    ) toIter [CloseHandle drop] eachStatic
-
-    1n32 terminate drop
-  ];
-
-  private INIT: [
-    STARTUPINFOW        !startupInfo
-    PROCESS_INFORMATION !processInfo
+  exitCode: [
+    out:;
+    errorTrait: @out processInformation.hProcess GetExitCodeProcess;
+    errorTrait 0 = ~
   ];
 }];
+
+# Start 'process' represented by 'command'
+# in:
+#   command
+# out:
+#   process
+#
+# Examples:
+#   1) "program" startProcess.closeInactive
+#
+#   2) "program programArgument" startProcess.closeInactive
+#
+#   3) program: "program" startProcess;
+#      program.operationSucceed [program.closeInactive] [
+#        ("Failed to start program: " program.operationError) printList
+#      ] if
+#
+#   4) program: "program" startProcess;
+#      program.tillActive
+#      exitCode: program.exitCode;
+#      program.close
+startProcess: [
+  command:;
+  {
+    process:            Process;
+    command:            command utf16;
+    operationErrorCode: Nat32;
+    toBeClosed:         Cond;
+
+    operationError: [operationErrorCode new];
```

Comment:
I think a property accessor name should start with `get` (methods like `size` on collections are an exception, because they are a part of a predefined interface). I suggest `getErrorCode`. In this case it would be good to also change the property name to `errorCode` and `updateOperationError` to `updateErrorCode`.